registration/include/pcl/registration/ndt.h

```diff
@@ -455,19 +495,19 @@ namespace pcl
         *
         * The precomputed angular derivatives for the jacobian of a transformation vector, Equation 6.19 [Magnusson 2009]. 
         */
-      Eigen::Matrix<double, 8, 4> angular_jacobian_;
+      Eigen::Matrix<float, 8, 4> angular_jacobian_;
```

Comment:
Does the change from `double` to `float` provide computational benefit? SSE intrinsics for double do exist

Reply:
When I wrote the code a few years ago, the float-version was significantly faster than the double-version on a bit old PC. But, I did bench-marking again today, and now the double-version was a bit faster than the float-version interestingly.

I guess this stems from AVX that has 256bit width and can process four doubles at a time. Indeed, I confirmed that the double-version got slower when AVX was disables.

I think AVX is very common now, and the use of float is not worthy. I replaced floats with doubles in the last commit.

(align 17k points 100 times)
```
Eigen::SimdInstructionSetsInUse(): AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2
float :17153[msec]
double:16844.1[msec]

Eigen::SimdInstructionSetsInUse(): SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2
float :17613[msec]
double:18299.2[msec]
````